blobfile/_context.py

```diff
@@ -776,6 +775,12 @@ def md5(self, path: RemoteOrLocalPath) -> str:
             with self.BlobFile(path, "rb") as f:
                 return common.block_md5(f).hex()
 
+    def last_version_seen(self, file: TextIO | BinaryIO) -> Optional[str]:
+        try:
+            return file.buffer.raw._version  # type: ignore
+        except AttributeError:
```

Comment:
hmm i'm not sure this is right, i think it depends on the mode. if we're offering a public api it should be fairly robust to that kind of thing

Reply:
You're right. How about this?